conjur/config.py

```diff
@@ -29,25 +31,28 @@ class Config():
     FIELDS = [
         ('conjur_account', 'account', True),
         ('conjur_url', 'url', True),
-        ('cert_file', 'ca_bundle', False),
+        ('cert_file', 'ca_bundle', True),
     ]
 
     _config = {}
 
     def __init__(self, config_file=DEFAULT_CONFIG_FILE):
         # pylint: disable=logging-fstring-interpolation
-        logging.debug(f"Fetching connection details from filesystem {config_file}")
 
+        logging.debug(f"Fetching connection details from filesystem {config_file}")
         config = None
         with open(config_file, 'r') as config_fp:
             config = load(config_fp, Loader=Loader)
 
-        for config_field_name, attribute_name, mandatory in self.FIELDS:
-            if mandatory:
-                assert config_field_name in config
+        if not config:
+            for config_field_name, attribute_name, mandatory in self.FIELDS:
+                if mandatory:
+                    assert config_field_name in config
```

Comment:
I would throw specific error here and a different in the setattr.
this way you can inform if a value is mandatory or not

Reply:
as in two catches in client.py (this calling class) without the `else raise ConfigurationMissingException`?

except AssertionError
except ConfigurationMissingException

```
if mandatory:
  assert config_field_name in config, "THIS IS MANDETORY" # raises assertionerror

try:
   setattr(self, attribute_name, config[config_field_name])
   self._config[attribute_name] = getattr(self, attribute_name) 
except ConfigurationMissingException
```

Reply:
Sorry I wasn't clear. i wouldn't raise assert but a dedicated exception like MissingParameterException.
Basically we should use assert to simplify verification specially in constructors [see here](https://wiki.python.org/moin/UsingAssertionsEffectively) but after that catch and throw a specific exception.